src/blog/understanding-ecmascript-part-2.md

```diff
@@ -0,0 +1,305 @@
+---
+title: 'Understanding the ECMAScript spec, part 2'
+author: '[Marja Hölttä](https://twitter.com/marjakh), speculative specification spectator'
+avatars:
+  - marja-holtta
+date: 2020-02-05 13:33:37
+tags:
+  - ECMAScript
+description: 'Tutorial on reading the ECMAScript specification'
+tweet: ''
+---
+
+... where we practice our awesome spec reading skills some more.
+
+## Previous episodes
+
+If you haven't had a look at the previous episodes, now it's a good time to do so!
+
+[Part 1](https://v8.dev/blog/understanding-ecmascript-part-1)
+
+## Ready for part 2?
+
+A fun way to get to know the spec is to start with a JavaScript feature we know is there, and find out how it's specified.
+
+We know that properties are looked up in the prototype chain: if an object doesn't have the property we're trying to read, we walk up the prototype chain until we find it (or find an object which no longer has a prototype).
+
+For example:
+
+```javascript
+const o1 = {'foo' : 2511};
+const o2 = {};
+console.log(o2.foo); // undefined
+o2.__proto__ = o1;
```

Comment:
Let's use `Object.setPrototypeOf` instead of `__proto__` (throughout the entire article). `__proto__` is a legacy web compat wart, and `Object.setPrototypeOf` is more readable.